### PR TITLE
Add tree view support for Route, Pod, PVC, PV

### PR DESCRIPTION
**Pull request by djwhatle**
When paired with https://github.com/konveyor/mig-controller/pull/1065, you get a really good view of the most important (error prone) resources in a migration.

## New Demo (updated Apr 16)
https://user-images.githubusercontent.com/7576968/115084465-f5dca480-9ed6-11eb-830f-50c8cdd65144.mp4

### Older Demos
**Direct Migration Demo**
https://www.youtube.com/watch?v=rrClHP8GQiM

**Indirect Migration Demo**
https://www.youtube.com/watch?v=hdbg87agznQ

### CODE REVIEW

**Comment by eriknelson** [comment]
Oh man, Derek this is super cool 👍

⁣Get BlueMail for Android ​

On Apr 9, 2021, 4:54 PM, at 4:54 PM, Derek Whatley ***@***.***> wrote:
***@***.*** this may need refactoring. I went with a brute-force

**Comment by djwhatle** [comment]
@ibolton336 any further concerns on styling? It seems like `yarn lint:fix` doesn't change anything, so I'm guessing you resolved the problems in your PR to my branch.

**Comment by ibolton336** [comment]
Everything is looking good on my end - just tested. Nice work getting the new states implemented.

**Review by ibolton336 — DISMISSED** [review]
LGTM

**Comment by vconzola** [comment]
@djwhatle Apparently, I need subtitles because without audio I couldn't tell what exactly the PR is doing? Is it just adding the terminating status to some of the resources? Also, not sure where the phase names came from at the top of the tree.

**Comment by djwhatle** [comment]
@vconzola this PR adds support for displaying status info for `Route, Pod, PVC, PV`

**Comment by vconzola** [comment]
@djwhatle OK, I think I see it now. Are those tags added to the beginning of the resource names?

**Comment by djwhatle** [comment]
@ibolton336 ready for final review.

**Comment by djwhatle** [comment]
I fixed the `oc get` button for all the new tree view resources.

File: src/app/debug/duck/selectors.ts [hunk 1]
```diff
@@ -34,65 +34,255 @@ const getResourceStatus = (debugRef: IDebugRefRes): IDerivedDebugStatusObject =>
...
+          null,
+          null,
+          null
+        ),
+      };
+    }
+    case 'PVC': {
+      const { phase } = debugRef.value.data.object.status;
+      const { deletionTimestamp } = debugRef.value.data.object.metadata;
+      const hasFailure = phase === 'Lost';
```
Comment (pranavgaikwad) [inline src/app/debug/duck/selectors.ts:98]: Interesting, didnt know PVC has 3 phases.

File: src/app/debug/duck/selectors.ts [hunk 2]
```diff
@@ -34,65 +34,255 @@ const getResourceStatus = (debugRef: IDebugRefRes): IDerivedDebugStatusObject =>
...
+      };
+    }
     case 'Backup': {
       const { errors, warnings, phase } = debugRef.value.data.object.status;
+      const { deletionTimestamp } = debugRef.value.data.object.metadata;
       const hasWarning = warnings?.length > 0 || phase === 'PartiallyFailed';
       const hasFailure = errors?.length > 0 || phase === 'Failed';
       const hasCompleted = phase === 'Completed';
       const hasRunning = phase === 'InProgress';
+      const hasTerminating = deletionTimestamp != undefined;
```
Comment (pranavgaikwad) [inline src/app/debug/duck/selectors.ts:181]: +1

**Review by pranavgaikwad — APPROVED** [review]
LGTM

**Review by ibolton336 — APPROVED** [review]
LGTM!

**Pull request merged**